Dependencies/xr/Source/ARCore/XR.cpp

```diff
@@ -664,16 +669,221 @@ namespace xr
                     float rawPose[7]{};
                     ArHitResult_getHitPose(xrContext->Session, hitResult, tempPose);
                     ArPose_getPoseRaw(xrContext->Session, tempPose, rawPose);
-                    HitResult hitResult{};
-                    RawToPose(rawPose, hitResult.Pose);
+                    HitResult hitTestResult{};
+                    RawToPose(rawPose, hitTestResult.Pose);
 
-                    hitResult.NativeTrackable = reinterpret_cast<NativeTrackablePtr>(trackable);
-                    filteredResults.push_back(hitResult);
+                    hitTestResult.NativeTrackable = reinterpret_cast<NativeTrackablePtr>(trackable);
+                    filteredResults.push_back(hitTestResult);
                     frameTrackables.push_back(trackable);
                 }
             }
         }
 
+        // Converts an image bitmap to grayscale assumes an RGB8 or RGB8A image for use with ARCore.
+        void ConvertBitmapToGrayscale(
+            const uint8_t* image_pixel_buffer,
+            int32_t width,
+            int32_t height,
+            int32_t stride,
+            uint8_t** out_grayscale_buffer)
```

Comment:
To me it would make more sense for the caller to allocate this memory and just pass in a pointer. Then they can manage the memory however they want (e.g. a `unique_ptr`).